datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py

```diff
@@ -212,4 +212,11 @@ def _make_counters(self, key=None, counter_data=([], ())):  # Key left in for re
 
     @classmethod
     def _no_instance(cls, inst_name):
```

Comment:
This is called here (and doesn't depend on the `check_instance` bool, since we're using an `and` here):

https://github.com/DataDog/integrations-core/blob/78e45f56ba745ad488c4ddbaaf2daf83f732abf2/datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py#L180-L182

The customer was passing this:

```yaml
additional_metrics:
  - ['Webdienst', none, 'Verbindungsversuche/s', iis.test.german, gauge]
```

So I assume `none` is treated as `null` by YAML, then `None` in Python, then it triggers the `AttributeError`.

But how come we don't see this exception given that most integrations pass `None` as the `inst_name` in their default counters is still a total mystery to me.

Reply:
Updated: looks like `none` is read as the string `"none"` by `pyyaml`, so the traceback in PR description is most likely not triggered by that.

Also from the traceback, the exception occurs during "refresh counters":

https://github.com/DataDog/integrations-core/blob/3db84e457a3ce6747867943131ed38a293911c87/iis/datadog_checks/iis/iis.py#L58-L61

And so that means that `IIS.__init__()` works just fine. 🤔

Reply:
Oh _right_, so in the case of default metrics programmatically added by integrations, or by tests such as [this one](https://github.com/DataDog/integrations-core/blob/e9be0e11ee34cc3fdfad665a412dcc3c3b7bac2d/datadog_checks_base/tests/test_pdhbasecheck.py#L42-L50), `._no_instance()` doesn't run because `check_instance` is `False`:

https://github.com/DataDog/integrations-core/blob/78e45f56ba745ad488c4ddbaaf2daf83f732abf2/datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py#L98

Contrary to the `additional_metrics` case which the customer is using, in which `check_instance` is `True` and so Python must evaluate the second part of the condition:

https://github.com/DataDog/integrations-core/blob/78e45f56ba745ad488c4ddbaaf2daf83f732abf2/datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py#L104

This explains why the bug this PR addresses doesn't manifest itself for default metrics.

Reply:
When the customer passes `additional_metrics` which contain `'none'` (the string) as the `inst_name`, the condition evaluates to `True` and sets `inst_name` to `None:

https://github.com/DataDog/integrations-core/blob/78e45f56ba745ad488c4ddbaaf2daf83f732abf2/datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py#L181-L183

Then if creating the counter fails, we register the counter into `_missing_counters`, and at this point we have `inst_name = False` and `check_instance = True` (because this comes from `additional_metrics`):

https://github.com/DataDog/integrations-core/blob/78e45f56ba745ad488c4ddbaaf2daf83f732abf2/datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py#L199-L204

Then the `iis` check calls `_make_counters()` during refresh:

https://github.com/DataDog/integrations-core/blob/3db84e457a3ce6747867943131ed38a293911c87/iis/datadog_checks/iis/iis.py#L58-L60

And _this time_ the second half of the condition is evaluated, because `check_instance` is `True`, but `inst_name` is `None` and so we run into this bug.

Nasty!